### PR TITLE
Support APIGroup in SAR/RAR

### LINKED ISSUE #5751: SAR/RAR does not allow checking API group

### PR DESCRIPTION
**Pull request by liggitt**
Fixes #5751

### CODE REVIEW

**Comment by deads2k** [comment]
Is this logically tied to all those commits?  I would expect it to be distinct.

**Comment by liggitt** [comment]
built on top of commits with integration tests that exposed the issue... last commit is the only interesting one

**Comment by liggitt** [comment]
[test]

**Comment by deads2k** [comment]
still WIP?

**Comment by liggitt** [comment]
Work-incomplete-but-not-in-progress

**Comment by liggitt** [comment]
rebased, at least

**Comment by deads2k** [comment]
ready for review?

**Comment by danmcp** [comment]
@liggitt Bump

**Comment by liggitt** [comment]
other than the API field names, this is ready for review

**Comment by jwforres** [comment]
re [test] (bindata issue fixed)

File: pkg/authorization/authorizer/attributes_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package authorizer
+
+import (
+	"testing"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+)
+
+func TestAuthorizationAttributes(t *testing.T) {
```
Comment (deads2k) [inline pkg/authorization/authorizer/attributes_test.go]: huh?

File: pkg/build/admission/strategyrestrictions/admission.go [hunk 2]
```diff
@@ -98,6 +98,7 @@ func (a *buildByStrategy) checkBuildAuthorization(build *buildapi.Build, attr ad
 	subjectAccessReview := &authorizationapi.LocalSubjectAccessReview{
 		Action: authorizationapi.AuthorizationAttributes{
 			Verb:         "create",
+			Group:        "",
```
Comment (deads2k) [inline pkg/build/admission/strategyrestrictions/admission.go]: use constant, looks like `buildapi.GroupName`

File: pkg/cmd/server/origin/ensure.go [hunk 3]
```diff
@@ -185,7 +185,7 @@ func (c *MasterConfig) ensureComponentAuthorizationRules() {
 	}
 
 	// Wait until the policy cache has caught up before continuing
-	review := &authorizationapi.SubjectAccessReview{Action: authorizationapi.AuthorizationAttributes{Verb: "get", Resource: "clusterpolicies"}}
+	review := &authorizationapi.SubjectAccessReview{Action: authorizationapi.AuthorizationAttributes{Verb: "get", Group: "", Resource: "clusterpolicies"}}
```
Comment (deads2k) [inline pkg/cmd/server/origin/ensure.go]: constant

File: test/util/policy.go [hunk 4]
```diff
@@ -16,8 +16,8 @@ const (
 
 // WaitForPolicyUpdate checks if the given client can perform the named verb and action.
 // If PolicyCachePollTimeout is reached without the expected condition matching, an error is returned
-func WaitForPolicyUpdate(c *client.Client, namespace, verb, resource string, allowed bool) error {
-	review := &authorizationapi.LocalSubjectAccessReview{Action: authorizationapi.AuthorizationAttributes{Verb: verb, Resource: resource}}
+func WaitForPolicyUpdate(c *client.Client, namespace, verb, resourceGroup, resource string, allowed bool) error {
```
Comment (deads2k) [inline test/util/policy.go]: If you want a `GroupResource`, take a `GroupResource`.

**Comment by deads2k** [comment]
minor comments.  lgtm even with the field names.

**Comment by liggitt** [comment]
@deads2k comments addressed, sanity check my api group references

**Comment by deads2k** [comment]
Verify failed.  @danmcp does the cause analysis happen before or after rosie comments on pulls?  It's a little annoying that I have to click through.

**Comment by danmcp** [comment]
@deads2k after

File: pkg/image/registry/imagestream/strategy.go [hunk 5]
```diff
@@ -411,6 +411,7 @@ func (v *TagVerifier) Verify(old, stream *api.ImageStream, user user.Info) field
 		subjectAccessReview := authorizationapi.SubjectAccessReview{
 			Action: authorizationapi.AuthorizationAttributes{
 				Verb:         "get",
+				Group:        api.GroupName,
```
Comment (deads2k) [inline pkg/image/registry/imagestream/strategy.go]: imageapi

Comment (deads2k) [inline pkg/image/registry/imagestream/strategy.go]: > imageapi

Oh, I see.  Its one of *those* packages.

**Comment by deads2k** [comment]
lgtm [merge]

**Pull request merged**